### ISSUE TITLE
jsonschema.RefResolver is deprecated

### ISSUE THREAD

**Issue by miettal** [python-restx/flask-restx#553]
### **Code**

```python
import flask_restx
```

### **Expected Behavior**

no error

### **Actual Behavior**

output below warning.
```
E   DeprecationWarning: jsonschema.RefResolver is deprecated as of v4.18.0, in favor of the https://github.com/python-jsonschema/referencing library, which provides more compliant referencing behavior as well as more flexible APIs for customization. A future release will remove RefResolver. Please file a feature request (on referencing) if you are missing an API for the kind of customization you need.
```

### **Environment**
- python 3.11
- Flask==2.3.2
- flask-restx==1.1.0
- jsonschema==4.18.3

### **Additional Context**

This is your last chance to provide any pertinent details, don't let this opportunity pass you by!

**Comment by miettal** [comment]
it seems jsonschema reference resolving function is sepalated from jsonschema package.
currently the function provided by referencing package.

https://github.com/python-jsonschema/referencing

**Comment by peter-doggart** [comment]
Thanks for the heads-up. Will likely have to fix `jsonscheme==4.17.3` until I can investigate the changes required to move to the new package.

**Comment by kannanprasanna** [comment]
@peter-doggart , could you please let us know if this issue will be addressed in any of the upcoming releases?

**Comment by peter-doggart** [comment]
Yes, it will be. I just haven't had an opportunity to dig into it yet. The primary issue is that flask-restx generates swagger 2.0 type docs that require JSON types compatible with JSON-Schema Draft 4, which is really, really old now.

**Comment by ske99ey** [comment]
Is there any idea as to when this will be resolved? Seems the above [PR](https://github.com/python-restx/flask-restx/pull/592) was opened but it basically failed all the test actions 😢

**Comment by golharam** [comment]
I'm seeing this too with the following configuration:
```
Flask==3.1.0
flask-restx==1.3.0
jsonschema==4.23.0
jsonschema-specifications==2024.10.1
```

**Comment by circulon** [comment]
I'm seeing it as well 
same as @golharam 

Any Idea when this might be addressed?

**Comment by dsrink** [comment]
@peter-doggart It would be great to see some progress on this. I would also be willing to throw a few hours of work on this. Is there something I can do to help?

**Comment by vivodi** [comment]
Any updates?

**Comment by peter-doggart** [comment]
@ske99ey @golharam @circulon @dsrink @vivodi I've pushed a patch to a PR, a bit of extra handling is required to create the same schema in the new library: https://github.com/python-restx/flask-restx/pull/639

I will try and get it merged and released this week, but would be great if any of you guys could check it out to make sure it's all working?

**Comment by vivodi** [comment]
I tested it, everything works fine.

**Comment by peter-doggart** [comment]
@ske99ey @vivodi Thanks for confirming so quickly, much appreciated.

I have merged everything and released v1.3.2 (not sure why the release pipeline decided all on its own to skip 1.3.1!). Please open a new issue and tag me if you find any problems.
